### PR TITLE
[WIP] New faster version of the RecordIO iterator

### PR DESCRIPTION
**Pull request by ptrendx**
Does not yet support shuffle (ignores that setting)

### CODE REVIEW

**Comment by ptrendx** [comment]
Test again?

**Comment by piiswrong** [comment]
@zhreshold

File: src/io/image_iter_common.h [hunk 1]
```diff
@@ -130,6 +132,9 @@ struct ImageRecParserParam : public dmlc::Parameter<ImageRecParserParam> {
     DMLC_DECLARE_FIELD(path_imgrec).set_default("")
         .describe("Path to the image RecordIO (.rec) file or a directory path. "\
                   "Created with tools/im2rec.py.");
+    DMLC_DECLARE_FIELD(path_imgidx).set_default("")
```
Comment (zhreshold) [inline src/io/image_iter_common.h]: Is `path_imgidx` required or optional?

Comment (ptrendx) [reply src/io/image_iter_common.h]: Optional.

**Comment by szha** [comment]
I can't seem to find the source code for TurboJPEG. Is it open source? I was able to find libjpeg-turbo, but it's [not the same thing as TurboJPEG](https://libjpeg-turbo.org/About/TurboJPEG)

**Comment by zhreshold** [comment]
@szha It's libjpegturbo, my bad
 https://github.com/libjpeg-turbo/libjpeg-turbo

**Comment by ptrendx** [comment]
Yes, it is libjpeg-turbo (https://github.com/libjpeg-turbo/libjpeg-turbo). It has 2 APIs though - libjpeg API and TurboJPEG API. I am using TurboJPEG API since it is more straightforward.

**Comment by szha** [comment]
Thanks. Would you update the name of the flag to reflect this, such as `USE_LIBJPEG_TURBO`?

**Comment by ptrendx** [comment]
Sure, I will do that.

**Comment by ptrendx** [comment]
Rebasing on current master.

**Comment by ptrendx** [comment]
@piiswrong It passed CI.

File: tools/rec2idx.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        if not self.is_open:
+            return
+        super(IndexCreator, self).close()
+        self.fidx.close()
+
+    def tell(self):
+        """Returns the current position of read head.
+        """
+        pos = ctypes.c_size_t()
+        check_call(_LIB.MXRecordIOReaderTell(self.handle, ctypes.byref(pos)))
```
Comment (piiswrong) [inline tools/rec2idx.py]: @ptrendx Please add this to recordio python API with a separate PR after release

**Comment by eric-haibin-lin** [comment]
Does it print warnings when shuffle is set to True?

**Comment by ptrendx** [comment]
Are you talking about the comment that shuffle does not work yet? This was when I was still working on it and is no longer true.

**Pull request merged**